### The following are steps in using this module
First import the classifier and numpy.

In [2]:
import SVMClassifier as SVMC
import numpy as np

Let's setup some data to feed into the SVM. A good day and an extremely bad day. The required features in order are: day length (hrs), average temperature (F), average humidity, max windspeed (mph), average windspeed (mph), max windgust (mph), precipitation (in). The first row is on a sunshine day, and the second row is a stormy day.

In [8]:
days = np.array([[11, 80, 80, 7, 3, 10, 0.], [8, 30, 50, 50, 30, 50, 1.]])
days.shape

(2, 7)

We see that days has the required shape to fit into SVMC. Great! Now to use it:

In [9]:
SVMC.predictOutage(days)

array([ 0.,  2.])

The classifier is telling us, for the sunny day, it's classifiying the day as 0, which is a case when there are only 0 - 2 outages. On the other hand, for the stormy day, it's classifying the day as 2, which is a case when there are 8 or more outages.

We can also predict the probabilities of each classification using the predictOutageProba function:

In [10]:
SVMC.predictOutageProba(days)

array([[ 0.96042868,  0.03666562,  0.0029057 ],
       [ 0.11566639,  0.39344899,  0.49088462]])

For the sunny day, there is a 96% chance that the day is of type 0, i.e. 0 - 2 outages.
For the stormy day, there is a 49% chance that the day is of type 2, i.e. more than 8 outages.

#### Caution:
The predictOutage function does not use the probabilities of the predictOutageProba function to make it's decision. The predictOutageProba function uses a pairwise correlation algorithm. Therefore they are not guaranteed to be consistent with each other, i.e. classification with the highest probability by predictOutageProba may not be the classification chosen by predictOutage. For more information please go to https://www.csie.ntu.edu.tw/~cjlin/libsvm/